services/transformer/features.go

```diff
@@ -0,0 +1,65 @@
+package transformer
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+
+	"github.com/rudderlabs/rudder-go-kit/logger"
+)
+
+const (
+	V0_SOURCE_TRANSFORMER_VERSION = "v0"
+	V1_SOURCE_TRANSFORMER_VERSION = "v1"
+)
+
+type FeaturesServiceConfig struct {
+	PollInterval             time.Duration
+	TransformerURL           string
+	FeaturesRetryMaxAttempts int
+}
+
+type FeaturesService interface {
+	SourceTransformerVersion() string
+	Get() json.RawMessage
```

Comment:
The service shouldn't expose the raw json message. Part of the abstraction's responsibility is to encapsulate this and provide higher level functions
```suggestion
	RouterTransform(destType string) bool
```